python/pyspark/sql/connect/session.py

```diff
@@ -427,7 +436,7 @@ def stop(self) -> None:
 
     @classmethod
     def getActiveSession(cls) -> Any:
-        raise NotImplementedError("getActiveSession() is not implemented.")
```

Comment:
Can we don't expose this but use `_active_spark_session` in ML instead? The behaviour of this implementation is different from regular Spark (e.g., 1. actions like `collect` should set the active Spark Session but this PR doesn't. 2. options provided should be set even when there's an active session)